plugins/outputs/http/http.go

```diff
@@ -0,0 +1,196 @@
+  ## Configure response header timeout in seconds. Default : 3
+  response_header_timeout = 3
+  ## Configure dial timeout in seconds. Default : 3
+  dial_timeout = 3
+
+  ## Data format to output.
+  ## Each data format has it's own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+  data_format = "influx"
+`
+
+const (
+	POST = "POST"
+
+	DEFAULT_RESPONSE_HEADER_TIMEOUT = 3
+	DEFAULT_DIAL_TIME_OUT           = 3
+)
+
+type Http struct {
+	// http required option
+	URL            string   `toml:"url"`
+	HttpHeaders    []string `toml:"http_headers"`
+	ExpStatusCodes []int    `toml:"expected_status_codes"`
+
+	// Option with http default value
+	ResHeaderTimeout int `toml:"response_header_timeout"`
+	DialTimeOut      int `toml:"dial_timeout"`
+
+	client     http.Client
+	serializer serializers.Serializer
+	// expStatusCode that stores option values received with expected_status_codes
+	expStatusCode map[int]bool
+}
+
+func (h *Http) SetSerializer(serializer serializers.Serializer) {
+	h.serializer = serializer
+}
+
+// Connect to the Output
+func (h *Http) Connect() error {
+	h.client = http.Client{
+		Transport: &http.Transport{
+			Dial: (&net.Dialer{
+				Timeout: time.Duration(h.DialTimeOut) * time.Second,
+			}).Dial,
+			ResponseHeaderTimeout: time.Duration(h.ResHeaderTimeout) * time.Second,
+		},
+	}
+
+	return nil
+}
+
+// Close is not implemented. Because http.Client not provided connection close policy. Instead, uses the response.Body.Close() pattern.
+func (h *Http) Close() error {
+	return nil
+}
+
+// Description A plugin that can transmit metrics over HTTP
+func (h *Http) Description() string {
+	return "A plugin that can transmit metrics over HTTP"
+}
+
+// SampleConfig provides sample example for developer
+func (h *Http) SampleConfig() string {
+	return sampleConfig
+}
+
+// Writes metrics over HTTP POST
+func (h *Http) Write(metrics []telegraf.Metric) error {
+	var mCount int
+	var reqBodyBuf []byte
+
+	for _, metric := range metrics {
+		buf, err := h.serializer.Serialize(metric)
+
+		if err != nil {
+			return fmt.Errorf("E! Error serializing some metrics: %s", err.Error())
+		}
+
+		reqBodyBuf = append(reqBodyBuf, buf...)
+		mCount++
+	}
+
+	reqBody, err := makeReqBody(h.serializer, reqBodyBuf, mCount)
+
+	if err != nil {
+		return fmt.Errorf("E! Error serialized metric is not assembled : %s", err.Error())
+	}
+
+	if err := h.write(reqBody); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (h *Http) write(reqBody []byte) error {
+	req, err := http.NewRequest(POST, h.URL, bytes.NewBuffer(reqBody))
+
+	for _, httpHeader := range h.HttpHeaders {
+		keyAndValue := strings.Split(httpHeader, ":")
+		req.Header.Set(keyAndValue[0], keyAndValue[1])
+	}
+
+	req.Close = true
+
+	res, err := h.client.Do(req)
+
+	if err := h.isOk(res, err); err != nil {
+		return err
+	}
+
+	res.Body.Close()
```

Comment:
If the status code does not match the expected set the function returns but the body is not closed.

Reply:
I have added below code
`defer resp.Body.Close()`